daft/expressions/expressions.py

```diff
@@ -819,7 +829,13 @@ def batch_func(self_series):
             name = name + "."
         name = name + getattr(func, "__qualname__")  # type: ignore[call-overload]
 
-        return StatelessUDF(name=name, func=batch_func, return_dtype=return_dtype, resource_request=None)(self)
+        return StatelessUDF(
+            name=name,
+            func=batch_func,
+            return_dtype=return_dtype,
+            resource_request=None,
+            batch_size=None,
```

Comment:
Let's make this configurable for `.apply`? We can also default to some reasonable number here.

This way `.apply` won't have to materialize the entire thing as a Python list before running, would be good for memory stability too.

`df["col"].apply(..., batch_size=8)`

Reply:
Not going to add this for now because it might mislead users into thinking we are going to be passing batches of data into the `.apply`.